server/conn.go

```diff
@@ -1119,16 +1113,42 @@ func (cc *clientConn) handleLoadData(ctx context.Context, loadDataInfo *executor
 			break
 		}
 	}
-	loadDataInfo.SetMessage()
-
 	if err != nil {
-		loadDataInfo.Ctx.StmtRollback()
+		logutil.Logger(ctx).Error("load data process stream error", zap.Error(err))
 	} else {
-		err = loadDataInfo.CheckAndInsertOneBatch(ctx)
-		if err == nil {
-			err = loadDataInfo.Ctx.StmtCommit()
+		err = loadDataInfo.EnqOneTask(ctx)
+		if err != nil {
+			logutil.Logger(ctx).Error("load data process stream error", zap.Error(err))
 		}
 	}
+}
+
+// handleLoadData does the additional work after processing the 'load data' query.
+// It sends client a file path, then reads the file content from client, inserts data into database.
+func (cc *clientConn) handleLoadData(ctx context.Context, loadDataInfo *executor.LoadDataInfo) error {
```

Comment:
how about finish the load data task inside the `LoadDataExec.Next()` function?

Reply:
this is done in another tmp closed #11670 to make load statement traceable, reopen it when this merged or many conflicts